collect_app/src/main/java/org/odk/collect/android/activities/GeoTraceOsmMapActivity.java

```diff
@@ -360,8 +360,11 @@ public void finish() {
     @Override
     protected void onDestroy() {
         super.onDestroy();
-    }
 
+        if (schedulerHandler != null && !schedulerHandler.isCancelled()) {
```

Comment:
As I said in the description the crash occurs only when we use OSM but we should cancel schedulerHandler here as well.